test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -3086,6 +3086,39 @@ private async Task DotnetRestore_WithFramework_SDKAnalysisLevelAndRestoreEnableP
             assetsFile.Targets[0].Libraries.Should().Contain(e => e.Name.Equals("Y"));
         }
 
+        [Theory]
+        [InlineData(null, "all")]
+        [InlineData("direct", "direct")]
+        [InlineData("all", "all")]
+        public void DotnetRestore_WritesExpectedAuditModeInAssetsFile(string userAuditMode, string expectedAuditMode)
```

Comment:
This test looks great. It is rebuilding a project for each iteration, but the cost is low and it ensures a new, separate project with each audit mode is tested.

nit: missing comments for Arrange/act/assert